packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -552,6 +566,13 @@ function createArrayNodeProxy(
 	allowAdditionalProperties: boolean,
 	customTargetObject?: object,
 ): TreeArrayNode {
+	if (customTargetObject !== undefined) {
+		// This causes `JSON.stringify` to give the same output as if the proxy were truly an array.
+		// Otherwise, it will stringify like an object (e.g. `{"0": ..., "1": ...}`) for a `customTargetObject`.
+		Reflect.defineProperty(customTargetObject, "toJSON", {
```

Comment:
I don't think this is a good approach. The whole point of customTargetObject is that the caller of this controls whats inherited. Having this code modify that target for every proxy created (ex: modify the schema object every time we create an instance of it) seems inefficient, inconsistent with how we handle other inherited properties, and potentially confusing as its very odd for this to modify the prototype.

Also I think if you make two array nodes, both with get the same inherited method, and thus toString the same. Ex: if you make one node then a second, and stringify the first is will report the contents of the second since you closed over the instance (proxy) rather than using the "this".

Reply:
Instead put this on the class as a method that gets passed into createArrayNodeProxy

Reply:
Yea, I wasn't sure the best way to do this. I've moved it to be a method on the class instead (which requires some annoying casts) - is that more appropriate?